opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -118,63 +118,118 @@ message MetricDescriptor {
+
+    // INT64 values are represents as signed 64-bit integers.
+    //
+    // A Metric of this Type MUST store its values as Int64DataPoint.
+    INT64 = 1;
+
+    // DOUBLE values are represents as double-precision floating-point
+    // numbers.
+    //
+    // A Metric of this Type MUST store its values as DoubleDataPoint.
+    DOUBLE = 2;
+
+    // Histogram measurement.
+    // Corresponding values are stored in HistogramDataPoint.
+    HISTOGRAM = 3;
 
     // Summary value. Some frameworks implemented Histograms as a summary of observations
     // (usually things like request durations and response sizes). While it
     // also provides a total count of observations and a sum of all observed
     // values, it calculates configurable percentiles over a sliding time
     // window.
     // Corresponding values are stored in SummaryDataPoint.
-    SUMMARY = 7;
+    SUMMARY = 4;
   }
+
+  // type is the type of values this metric has.
   Type type = 4;
 
+  // Temporality is the temporal quality values of a metric have. It
+  // describes how those values relate to the time interval over which they
+  // are reported.
+  enum Temporality {
+    // INVALID_TEMPORALITY is the default Temporality, it MUST not be
+    // used.
+    INVALID_TEMPORALITY = 0;
+
+    // INSTANTANEOUS is a metric whose values are measured at a particular
+    // instant. The values are not aggregated over any time interval and are
+    // unique per timestamp. As such, these metrics are not expected to have
+    // an associated start time.
+    INSTANTANEOUS = 1;
+
+    // DELTA is a metric whose values are the aggregation of measurements
+    // made over a time interval. Successive metrics contain aggregation of
+    // values from continuous and non-overlapping intervals.
+    //
+    // The values for a DELTA metric are based only on the time interval
+    // associated with one measurement cycle. There is no dependency on
+    // previous measurements like is the case for CUMULATIVE metrics.
+    //
+    // For example, consider a system measuring the number of requests that
+    // it receives and reports the sum of these requests every second as a
+    // DELTA metric:
+    //
+    //   1. The system starts receiving at time=t_0.
+    //   2. A request is received, the system measures 1 request.
+    //   3. A request is received, the system measures 1 request.
+    //   4. A request is received, the system measures 1 request.
+    //   5. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_0 to
+    //      t_0+1 with a value of 3.
+    //   6. A request is received, the system measures 1 request.
+    //   7. A request is received, the system measures 1 request.
+    //   8. The 1 second collection cycle ends. A metric is exported for the
+    //      number of requests received over the interval of time t_0+1 to
+    //      t_0+2 with a value of 2.
+    DELTA = 2;
```

Comment:
how is the interval represented when reporting delta? Is it by using `start_time_unix_nano` and `time_unix_nano` fields in `Int64DataPoint` etc, ?

Reply:
That is the plan. Planning to address this issue in a following PR (https://github.com/MrAlias/opentelemetry-proto/commit/5cfae029570935d54ee1b1c903c31a5fb074c44e).

The ask has been to split up https://github.com/open-telemetry/opentelemetry-proto/pull/137 into smaller changes. The assumption that there would be breaking changes or inconsistencies between PRs was something planned and accepted.